src/test/template-analyzer_test.ts

```diff
@@ -0,0 +1,413 @@
+// /* eslint-env mocha */
+import * as assert from 'assert';
+import treeAdapter = require('parse5-htmlparser2-tree-adapter');
+import {TemplateAnalyzer} from '../template-analyzer';
+// @ts-ignore
+import {parse} from 'babel-eslint';
+
+describe('TemplateAnalyzer', () => {
+  describe('create', () => {
+    it('should reuse the analyzer for a given TaggedTemplateExpression', () => {
+      const ast = parse('html`<h1>Hello</h1>`');
+      const taggedTemplate = ast.body[0].expression;
+      const analyzer1 = TemplateAnalyzer.create(taggedTemplate);
+      const analyzer2 = TemplateAnalyzer.create(taggedTemplate);
+      assert.strictEqual(analyzer1, analyzer2);
+    });
+    it('should handle parse errors', () => {
+      const ast = parse('html`<h1>Hello<///h1>>>>`');
+      const taggedTemplate = ast.body[0].expression;
+      const analyzer = TemplateAnalyzer.create(taggedTemplate);
+      assert.deepEqual(analyzer.errors, [
+        {
+          code: 'invalid-first-character-of-tag-name',
+          endCol: 12,
+          endLine: 1,
+          endOffset: 11,
+          startCol: 12,
+          startLine: 1,
+          startOffset: 11
+        }
+      ]);
+    });
+  });
+  describe('getLocationFor', () => {
+    it('should return the location of the starting tag of an element', () => {
```

Comment:
add a `done` callback to these tests and call it after your assertions inside the visitor below

this way if the traverse function ever breaks, we'll know as the tests will never call done